### ISSUE TITLE
Dependent on global babel polyfill

### ISSUE THREAD

**Issue by kpdecker** [mozilla-b2g/mozilla-download#33]
This library utilizes babel's polyfill implementation which can lead to version conflict errors within a process. If possible, this should be compiled using babel's runtime mode to ensure module isolation.

**Comment by gaye** [comment]
Version conflict errors?

**Comment by kpdecker** [comment]
I had one version of babel polyfill that I attempted to load to run the
library but it failed as it was a different version that this code was
built against. Ending up having to do the very ugly
require('mozilla-download/node_modules/babel/polyfill') call to make it
work properly.

Outside of loading the wrong version and finally figuring out the cryptic
error, I can't load your library in the same process as the rest of my
build chain since it uses traceur, which also uses its own version of the
global polyfill which will conflict.

Basically using globals and DLL hell ensues.
